### PR TITLE
🔒 Amend `p256` Test Coverage

### PR DESCRIPTION
**Pull request by pcaversaccio**
### 🕓 Changelog

In the test function `testVerifyWycheproofData` for the `p256` contract, we flip the `s` parameter if it is higher than the malleability threshold to validate the correctness of the underlying implementation (h/t @cairoeth for raising this point). We also `bound` the fuzzed private key in the `testFuzzVerifyWithValidSignature` test to account for the secp256r1 curve specifications. Finally, we refactor (i.e. remove) the `_SIGNATURE_INCREMENT` constant in the `ecdsa` contract by using the built-in Vyper constant `max_value(int256)` instead. You can validate the equivalence of the two constants by using [`titanoboa`](https://github.com/vyperlang/titanoboa):

```console
~$ pip install git+https://github.com/vyperlang/titanoboa@vyper-0.4.0
~$ python
>>> import boa
>>> boa.eval("convert(max_value(int256), uint256)") == 57_896_044_618_658_097_711_785_492_504_343_953_926_634_992_332_820_282_019_728_792_003_956_564_819_967
True
```

#### 🐶 Cute Animal Picture

image

### CODE REVIEW

File: test/utils/P256.t.sol [hunk 1]
```diff
@@ -110,9 +112,18 @@ contract P256Test is Test {
...
+            if (s <= _N) {
                 assertEq(
-                    P256.verify_sig(hash, r, s, x, y),
+                    P256.verify_sig(
+                        hash,
+                        r,
+                        /**
+                         * @dev Flip the `s` parameter if it is higher than the malleability threshold.
+                         */
+                        (s > _MALLEABILITY_THRESHOLD) ? (_N - s) : s,
```
Comment (cairoeth) [inline test/utils/P256.t.sol:123]: FYI there's a case (#1) that tests for signature malleability -- I imagine that by flipping the s param for all, it cancels that test

File: test/utils/P256.t.sol [hunk 2]
```diff
@@ -133,6 +144,7 @@ contract P256Test is Test {
         string calldata message
     ) public {
         (, uint256 key) = makeAddrAndKey(signer);
+        key = bound(key, 1, _N - 1);
```
Comment (cairoeth) [inline test/utils/P256.t.sol:147]: maybe instead just get the uint256 from fuzzing and bound it, without needing `makeAddrAndKey`

File: test/utils/P256.t.sol [hunk 3]
```diff
@@ -14,6 +14,8 @@ contract P256Test is Test {
     using BytesLib for bytes;
     using stdJson for string;
 
+    uint256 private constant _N =
```
Comment (cairoeth) [inline test/utils/P256.t.sol:17]: maybe u can pull this from the contract instead?

**Review by cairoeth — COMMENTED** [review]
nits

File: test/utils/P256.t.sol (hunk 3 above)
Comment (pcaversaccio) [reply test/utils/P256.t.sol:17]: It's an `internal` `constant` in my `p256` contract and thus there is no getter for it.

File: test/utils/P256.t.sol (hunk 1 above)
Comment (pcaversaccio) [reply test/utils/P256.t.sol:123]: I have a specific malleability test [here](https://github.com/pcaversaccio/snekmate/blob/main/test/utils/P256.t.sol#L125-L129). So I think this should be fine.

**Pull request merged**